libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/events/EventsTestHelper.kt

```diff
@@ -0,0 +1,466 @@
+                "stepCount value",
+                stepCount.toLong(),
+                content["stepCount"]!!.contents
+            )
+            assertEquals(
+                "originalStepCount value",
+                originalStepCount.toLong(),
+                content["originalStepCount"]!!.contents
+            )
+            assertEquals(
+                "legIndex value",
+                legIndex.toLong(),
+                content["legIndex"]!!.contents
+            )
+            assertEquals(
+                "legCount value",
+                legCount.toLong(),
+                content["legCount"]!!.contents
+            )
+            assertEquals(
+                "stepIndex value",
+                stepIndex.toLong(),
+                content["stepIndex"]!!.contents
+            )
+            assertEquals(
+                "voiceIndex value",
+                voiceIndex.toLong(),
+                content["voiceIndex"]!!.contents
+            )
+            assertEquals(
+                "bannerIndex value",
+                bannerIndex.toLong(),
+                content["bannerIndex"]!!.contents
+            )
+            assertEquals(
+                "totalStepCount value",
+                totalStepCount.toLong(),
+                content["totalStepCount"]!!.contents
+            )
+            assertTrue(content.containsKey("appMetadata"))
+            content["appMetadata"]!!.verifyAppMetadata(appMetadata)
+        }
+    }
+
+    fun Value.verifyAppMetadata(appMetadata: AppMetadata) {
+        (contents as Map<String, Value>).let { appMetadataContent ->
+            assertEquals(
+                "appMetadata, name value",
+                appMetadata.name,
+                appMetadataContent["name"]!!.contents
+            )
+            assertEquals(
+                "appMetadata, version value",
+                appMetadata.version,
+                appMetadataContent["version"]!!.contents
+            )
+            assertEquals(
+                "appMetadata, sessionId value",
+                appMetadata.sessionId,
+                appMetadataContent["sessionId"]!!.contents
+            )
+            assertEquals(
+                "appMetadata, userId value",
+                appMetadata.userId,
+                appMetadataContent["userId"]!!.contents
+            )
+        }
+    }
+
+    fun Value.verifyTelemetryLocation(location: TelemetryLocation) {
+        (contents as Map<String, Value>).let { content ->
+            assertEquals(location.latitude, content["lat"]!!.contents)
+            assertEquals(location.longitude, content["lng"]!!.contents)
+            assertEquals(location.speed.toDouble(), content["speed"]!!.contents)
+            assertEquals(location.bearing.toDouble(), content["course"]!!.contents)
+            assertEquals(location.altitude, content["altitude"]!!.contents)
+            assertEquals(location.timestamp, content["timestamp"]!!.contents)
+            assertEquals(
```

Comment:
Shouldn't you use tolerance for double here and everywhere else?

Reply:
these values are accurate, why do you think we need tolerance?

Reply:
Because double comparison without tolerance is deprecated and will fail:

```
    @Deprecated
    public static void assertEquals(String message, double expected,
            double actual) {
        fail("Use assertEquals(expected, actual, delta) to compare floating-point numbers");
    }
```

In your case it's Object though, not double. Then it won't fail I guess. But if you did any calculations under the hood, you might have gotten errors like "expected 22.0, actual 22.000000001".